ddp-workspace/projects/study-builder-ui/src/app/components/text-question-block/text-question-block.component.ts

```diff
@@ -13,35 +14,83 @@ import { SimpleTemplate } from '../../model/core-extended/simpleTemplate';
     styleUrls: ['./text-question-block.component.scss'],
     changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class TextQuestionBlockComponent implements OnInit {
-
-    @Input()
-    definitionBlock$: Observable<QuestionBlockDef<TextQuestionDef>>;
+export class TextQuestionBlockComponent implements OnInit, OnDestroy {
+    @Input() definitionBlock$: Observable<QuestionBlockDef<TextQuestionDef>>;
     angularClientBlock$: Observable<ActivityTextQuestionBlock>;
+    validationErrorMessages: string[] = [];
+    private blockCurrentAnswer = '';
+    private sub = new Subscription();
 
-    constructor(private config: ConfigurationService) {
-    }
-
+    constructor(
+        private config: ConfigurationService,
+        private questionConverter: ActivityQuestionConverter
+    ) {}
 
     ngOnInit(): void {
         this.angularClientBlock$ = this.definitionBlock$.pipe(
             tap(defBlock => console.log('getting defblock: %o', defBlock)),
             filter(block => !!block),
-            map(defBlock => this.buildFromDef(defBlock)));
+            map(defBlock => this.buildFromDef(defBlock)),
+            tap((block: ActivityTextQuestionBlock) => this.validate(block))
+        );
+    }
+
+    ngOnDestroy(): void {
+        this.sub.unsubscribe();
+    }
+
+    valueChanged(value: string): void {
+        this.blockCurrentAnswer = value;
+        const valueChangedSub = this.angularClientBlock$.subscribe((block: ActivityTextQuestionBlock) => this.validate(block, value));
+        this.sub.add(valueChangedSub);
```

Comment:
We might be subscribing every time a value is updated in the input field? Consider refactoring code so that this subscription only has to occur once. Or that it is not needed.

Reply:
You are right. 
Fixed.